### PR TITLE
fix(products): stop preview metering 500s and clarify error states

### PR DESCRIPTION
**Pull request by lvndry**
## What this PR does

Product pages like `/products/tiktok` were failing in the browser with a misleading **“Product Not Found”** message even though the product exists. The root cause was the backend crashing with **500** on anonymous overview requests due to an unsupported MongoDB `findAndModify.set_on_insert` parameter in preview usage metering.

This PR fixes the MongoDB upsert logic, returns **429** when preview/monthly limits are hit, **503** when metering itself fails, and updates the frontend to show a dedicated **“Couldn’t load this report”** state on 5xx instead of treating it as a missing product.

## Why

- **Users:** Anonymous visitors and free-tier users could not load product reports at all; Pro users were unaffected only when auth forwarded correctly.
- **Operators:** Production `/api/products/{slug}/overview` was returning 500 for every anonymous request with a preview token.

## How product page behaviour changes

| Path | Change |
|---|---|
| Anonymous GET `/products/{slug}/overview` | No longer 500 from preview metering; succeeds when under limit |
| Anonymous preview limit exceeded | Returns **429** (was crashing or inconsistent) |
| Preview metering MongoDB failure | Returns **503** instead of unhandled 500 |
| Signed-in Pro users | Still bypass preview/monthly caps (unchanged) |
| Frontend client refetch on 5xx | Shows server-error UI with retry, not “Product Not Found” |
| Frontend client refetch on 429 | Shows existing usage-limit UI (unchanged) |
| True 404 product | Still shows “Product Not Found” (unchanged) |

## UX changes operators will notice

- Product pages that previously showed “Product Not Found” after a backend 500 now show **“Couldn’t load this report”** with a **Try again** action.
- Usage limit messaging is unchanged; limit responses use HTTP **429**.

## Developer-visible API changes

| Surface | Old | New |
|---|---|---|
| Preview/monthly quota exhausted | 423 (intended) / 500 (production bug) | **429** |
| Preview metering failure | 500 | **503** |

## Tests

- Backend: `product_preview_usage_test.py`, `tier_access_test.py` — 27 tests passing.
- Frontend: `product-page-fetch-state.test.ts`, `product-page-client.test.tsx` — 9 tests passing.

### Edge cases to verify in a staging session

1. Open `/products/tiktok` logged out — expected: report loads (200 overview).
2. Exhaust anonymous preview quota — expected: **429** and usage-limit UI, not 404.
3. Simulate backend 500 on overview — expected: “Couldn’t load this report”, not “Product Not Found”.
4. Open same product as Pro user — expected: no preview limit, full access.

## Notes for reviewers

- The production bug was `find_one_and_update(..., set_on_insert=...)` which Atlas rejects; the fix uses standard `$setOnInsert` / `$inc` update documents instead.
- `limit_reached` is checked before `server_error` in fetch-state derivation so a 429 on any parallel request still wins over a 503 on overview.

### CODE REVIEW

File: packages/backend/src/services/product_preview_usage.py [hunk 1]
```diff
@@ -107,40 +40,52 @@ async def check_and_increment(
...
+            {"$inc": {"count": 1}, "$set": {"last_seen": now}},
+            return_document=ReturnDocument.AFTER,
+        )
+        if updated:
+            return True, int(updated["count"])
 
-        return True, self._count_after_increment(before, month_key)
+        doc = await collection.find_one(key_filter, {"count": 1, "month_key": 1})
+        current = self._effective_count(doc, month_key)
+        return False, current
```
[the remark this answers is not in the document]
Comment (lvndry) [reply packages/backend/src/services/product_preview_usage.py]: Addressed in 40a8aa4: the hot path now tries an optimistic `find_one_and_update` increment first (1 roundtrip for returning users under limit). Fallback handles new docs (upsert), month rollover, and at-limit reads. Tests updated accordingly.

File: packages/frontend/app/(dashboard)/products/[slug]/product-page-client.tsx [hunk 2]
```diff
@@ -922,27 +942,87 @@ export default function CompanyPage({
...
         >
-          <ArrowLeft className="h-3.5 w-3.5" />
-          Browse Products
-        </Link>
+          <RotateCcw className="mr-2 h-3.5 w-3.5" />
+          Try again
+        </Button>
       </div>
     );
   }
```
[the remark this answers is not in the document]
Comment (lvndry) [reply packages/frontend/app/(dashboard)/products/[slug]/product-page-client.tsx]: Addressed in 40a8aa4: removed the explicit `server_error` JSX block and consolidated on the default fallback, keeping the richer copy plus Try again + Browse Products actions for server errors and other unexpected `!data` states.
